backend/app/routers/authentication.py

```diff
@@ -116,13 +117,20 @@ async def get_admin(
 
 
 @router.get("/users/me/admin_mode")
-async def get_admin_mode(current_username=Depends(get_current_user)) -> bool:
+async def get_admin_mode(
+    superadmin: bool = False, current_username=Depends(get_current_user)
+) -> bool:
     """Get Admin mode from User Object."""
     if (
         current_user := await UserDB.find_one(UserDB.email == current_username.email)
     ) is not None:
-        if current_user.admin_mode is not None:
-            return current_user.admin_mode
+        if current_user.admin:
```

Comment:
I'm a little bit confused about the use case. I thought admin_mode is the equivalent of the flag "superadmin"? Would it be possible you share an example of why it is needed here?
Thanks!

Reply:
we want to be able to pass in a flag to endpoints "superadmin=true" that grants you admin permissions just for that call. we dont want to micromanage turning admin_mode on/off for every call, but the dev team didn't like the idea of leaving it turned on for a user all the time either.

Reply:
Ah I understand now. Thank you for explanation!!